### DISCUSSION TITLE
Leaf to Root Node

### DISCUSSION THREAD

**Discussion by gthiet** [dowjones/react-dropdown-tree-select#471] [Q&A]
Given a leaf node, Is there a way to find it's full path to the root node?

**Comment by mrchief** [comment]
@gthiet Absolutely!

The `onChange/onNodeToggle` events bubbles up few useful properties such as `_depth`, `_id`, `_parent` (and `_children` for non-leaf nodes) which you can use to recurse up.

![image](https://user-images.githubusercontent.com/781818/35543539-a545aeda-0533-11e8-8bf0-d81bc154d154.png)

You can also assign "object paths" to your nodes which are then bubbled up with `onChange/onNodeToggle`. I do this in one of my projects to easily pick the node in the tree:

```jsx

// create a "path" property with its path from root
// this path can be used by something like https://github.com/mariocasciaro/object-path to retrieve the node later

const assignObjectPaths = (obj, stack) => {
  Object.keys(obj).forEach(k => {
    const node = obj[k]
    if (typeof node === 'object') {
      node.path = stack ? `${stack}.${k}` : k
      assignObjectPaths(node, node.path)
    }
  })
}

// somewhere before render
assignObjectPaths(data)

// render
<DropdownTreeSelect data={data} onChange={onChange} />
```

See a demo here:  [![Edit Tree node paths with React Dropdown Tree Select](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/l765q6lmrq)

**Comment by gthiet** [comment]
I ended up using the _id and depth properties to recursively grab the label property, thanks!
